cloudinit/net/cmdline.py

```diff
@@ -233,34 +235,31 @@ def read_initramfs_config():
     return None
 
 
-def _decomp_gzip(blob, strict=True):
-    # decompress blob. raise exception if not compressed unless strict=False.
+def _decomp_gzip(blob):
+    # decompress blob or return original blob
     with io.BytesIO(blob) as iobuf:
         gzfp = None
         try:
             gzfp = gzip.GzipFile(mode="rb", fileobj=iobuf)
             return gzfp.read()
         except IOError:
-            if strict:
-                raise
             return blob
         finally:
             if gzfp:
                 gzfp.close()
 
 
-def _b64dgz(b64str, gzipped="try"):
-    # decode a base64 string.  If gzipped is true, transparently uncompresss
-    # if gzipped is 'try', then try gunzip, returning the original on fail.
-    try:
-        blob = base64.b64decode(b64str)
-    except TypeError:
-        raise ValueError("Invalid base64 text: %s" % b64str)
+def _b64dgz(data):
+    """Decode a base64 string if encoded, if gzipped transparently uncompress
 
-    if not gzipped:
-        return blob
+    :return decompressed unencoded string of the data
+    """
+    try:
+        blob = base64.b64decode(data)
+    except (TypeError, ValueError):
+        return data
```

Comment:
Let's raise the error here as we *do* expect network-config payload to be b64 encoded.

Reply:
+1